server/src/service-layer/controllers/PaymentController.ts

```diff
@@ -176,4 +184,182 @@ export class PaymentController extends Controller {
       return { error: "Something went wrong" }
     }
   }
+
+  @SuccessResponse("200", "Created booking checkout session")
+  @Security("jwt")
+  @Post("booking")
+  public async getBookingPayment(
+    @Request() request: SelfRequestModel,
+    @Body() requestBody: UserBookingRequestingModel
+  ): Promise<BookingPaymentResponse> {
+    const { uid, customClaims } = request.user
+    const { startDate, endDate } = requestBody
+    if (!customClaims || !customClaims[AuthServiceClaims.MEMBER]) {
+      this.setStatus(403)
+      return {
+        message: "Need a membership to create bookings"
+      }
+    }
+    // Need to validate the booking date through a startDate and endDate range.
+    const earliestDate = new Date()
+    earliestDate.setUTCHours(0, 0, 0, 0)
+    const latestDate = new Date(earliestDate)
+    latestDate.setFullYear(earliestDate.getFullYear() + 1)
+    // The request start and end dates
+    if (
+      endDate.seconds < startDate.seconds ||
+      startDate.seconds * 1000 < earliestDate.getTime() ||
+      endDate.seconds * 1000 > latestDate.getTime()
+    ) {
+      this.setStatus(401)
+      return {
+        error:
+          "Invalid date, booking start date and end date must be in the range of today up to a year later. "
+      }
+    }
+    // Calculate number of days of booking, returns a 0 indexed number
+    function getNumberOfDays(startDate: Timestamp, endDate: Timestamp): number {
+      const MILLIS_IN_A_DAY = 1000 * 60 * 60 * 24
+      const numberOfDays =
+        (startDate.seconds * 1000 - endDate.seconds * 1000) / MILLIS_IN_A_DAY
+      return numberOfDays + 1
+    }
+    // Calculate number of days of booking
+    const numberOfDays = getNumberOfDays(startDate, endDate)
+    // Validate number of dates to avoid kiddies from forging bookings
+    if (numberOfDays > 10) {
+      this.setStatus(401)
+      return {
+        error: "Invalid date range, booking must be a maximum of 10 days. "
+      }
+    }
```

Comment:
we should use the function `datesToDateRange` from `DateUtils` for simplicity. (just need to merge in master)